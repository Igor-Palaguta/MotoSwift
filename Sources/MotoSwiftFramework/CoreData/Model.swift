import Foundation

public struct Model {
   public let entities: [Entity]
   public let index: [String: Entity]

   init(_ entities: [Entity]) {
      self.entities = entities
      self.index = Dictionary(uniqueKeysWithValues: entities.map { ($0.name, $0) })
   }
}
