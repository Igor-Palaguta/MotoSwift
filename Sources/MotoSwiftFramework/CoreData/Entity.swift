import Foundation

public struct Entity {
   public let name: String
   public let className: String?
   public let attributes: [Attribute]
   public let relationships: [Relationship]
   public let fetchedProperties: [FetchedProperty]
   public let parentEntityName: String?
   public let userInfo: [String: String]
}
