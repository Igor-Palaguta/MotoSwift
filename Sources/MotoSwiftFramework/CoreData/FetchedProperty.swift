import Foundation

public struct FetchedProperty: Equatable {
   public let name: String
   public let entityName: String
   public let predicateString: String
   public let userInfo: [String: String]
}
