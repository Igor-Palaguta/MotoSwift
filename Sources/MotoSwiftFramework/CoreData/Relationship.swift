import Foundation

public struct Relationship: Equatable {
    public let name: String
    public let entityName: String
    public let isOptional: Bool
    public let toMany: Bool
    public let isOrdered: Bool
    public let userInfo: [String: String]
}
