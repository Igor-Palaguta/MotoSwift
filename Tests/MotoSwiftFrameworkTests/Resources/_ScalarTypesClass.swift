import Foundation
import CoreData

public enum ScalarTypesClassAttributes: String {
   case boolean
   case double
   case float
   case int16
   case int32
   case int64
}

public enum ScalarTypesClassRelationships: String {
   case numerics
}

public enum ScalarTypesClassFetchedProperties: String {
   case eq_true
   case gt_100

   public var predicateString: String {
      switch self {
      case .eq_true:
         return "boolean == YES"
      case .gt_100:
         return "int16 > 100"
      }
   }

   public var entityName: String {
      switch self {
      case .eq_true:
         return "ScalarTypes"
      case .gt_100:
         return "ScalarTypes"
      }
   }
}

public class _ScalarTypesClass: NSManagedObject {
   public class var entityName: String {
      return "ScalarTypes"
   }

   @NSManaged public var boolean: Bool
   @NSManaged public var double: Double
   @NSManaged public var float: Float
   @NSManaged public var int16: Int16
   @NSManaged public var int32: Int32
   @NSManaged public var int64: Int64

   @NSManaged public var numerics: NumericTypesClass
}
