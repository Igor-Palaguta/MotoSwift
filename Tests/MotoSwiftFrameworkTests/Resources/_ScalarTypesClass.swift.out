// DO NOT EDIT. This file is machine-generated and constantly overwritten.
// Make changes to machine.stencil instead.

import Foundation
import CoreData

// swiftlint:disable file_length
// swiftlint:disable type_body_length
public enum ScalarTypesClassAttributes: String {
   case boolean
   case double
   case float
   case int16
   case int32
   case int64
}

public enum ScalarTypesClassRelationships: String {
   case numerics
}

public enum ScalarTypesClassFetchedProperties: String {
   case eqTrue
   case gt100

   public var predicateString: String {
      switch self {
      case .eqTrue:
         return "boolean == YES"
      case .gt100:
         return "int16 > 100"
      }
   }

   public var entityName: String {
      switch self {
      case .eqTrue:
         return "ScalarTypes"
      case .gt100:
         return "ScalarTypes"
      }
   }
}

// swiftlint:disable type_name
public class _ScalarTypesClass: NSManagedObject {
   public class var entityName: String {
      return "ScalarTypes"
   }

   @NSManaged public var boolean: Bool
   @NSManaged public var double: Double
   @NSManaged public var float: Float
   @NSManaged public var int16: Int16
   @NSManaged public var int32: Int32
   @NSManaged public var int64: Int64

   @NSManaged public var numerics: NumericTypesClass
}

